src/cryptography/hazmat/backends/openssl/backend.py

```diff
@@ -346,6 +347,7 @@ def _int_to_bn(self, num, bn=None):
         ownership of the object). Be sure to register it for GC if it will
         be discarded after use.
         """
+        assert (bn is None) or (bn is not None and bn != self._ffi.NULL)
```

Comment:
`bn = None` means that `bn` is immediately set to `NULL`. However if a non-None value is provided then write the value to that instead.

This assert removes the ambiguity of the `_int_to_bn(0, NULL)` vs `_int_to_bn(0)`.